plat/common/lcpu.c

```diff
@@ -59,7 +59,7 @@
  * for non-present CPUs and does not force us to configure the maximum number
  * of CPUs beforehand.
  */
-struct lcpu lcpus[CONFIG_UKPLAT_LCPU_MAXCOUNT];
+UKPLAT_PER_LCPU_DEFINE(struct lcpu, lcpus);
```

Comment:
That's good, yet I noticed that the `lcpus` global is referenced directly in https://github.com/unikraft/unikraft/blob/staging/plat/kvm/x86/lcpu_start.S#L312. Since we are abstracting the implementation away here, it would be proper to also get a reference to the current `lcpu` via an accessor there too.

Reply:
Agreed, I will add assembly macro's in the header :ok_hand: